e2e/docker/regressions.go

```diff
@@ -256,3 +257,34 @@ func (c ctx) issue1528(t *testing.T) {
 		})
 	}
 }
+
+// https://github.com/sylabs/singularity/issues/1586
+// In OCI mode, ensure that nothing is left in TMPDIR from a docker:// image with restrictive file permissions.
+func (c ctx) issue1586(t *testing.T) {
+	tmpDir, cleanup := e2e.MakeTempDir(t, c.env.TestDir, "issue1586-", "")
+	t.Cleanup(func() {
+		if !t.Failed() {
+			cleanup(t)
+		}
+	})
+
+	c.env.RunSingularity(
+		t,
+		e2e.WithProfile(e2e.OCIUserProfile),
+		e2e.WithCommand("exec"),
+		e2e.WithArgs("docker://almalinux:9.1-minimal-20230407", "/bin/true"),
+		e2e.WithEnv(append(os.Environ(), "TMPDIR="+tmpDir)),
+		e2e.ExpectExit(0,
+			e2e.ExpectError(e2e.UnwantedContainMatch, "permission denied"),
+		),
+	)
+
+	d, err := os.Open(tmpDir)
```

Comment:
This is more of a question than a comment, but I wonder what are the pros and cons of putting this code directly in the body of the `issue1586()`, as opposed to wrapping it as an `e2e.PostRun` passed as part of `RunSingularity()`...?

Reply:
I generally prefer not to use `e2e.PostRun` unless it is necessary, just beause I don't really like adding anonymous functions and more nesting / brackets when it's not necessary. It's just a personal preference.

There is a specific, and fairly common, situation where `e2e.PostRun` is required....

Often we are calling `RunSingularity` with `AsSubtest`, so that e.g. when looping over different profiles, each `singularity` executions using a different profile takes place in a different subtest.

The subtest is created within `RunSingularity`, and is not accessible outside... so if we want to do any checks, we must do them, and fail the subtest, we have to do it in `PostRun`, where the `t` passed in is the subtest.